### PR TITLE
line-awesome: init at 1.3.0

### PR DESCRIPTION
**Pull request by puzzlewolf**
###### Motivation for this change
Line Awesome is a free alternative to Font Awesome, in line icon style. 
https://icons8.com/line-awesome/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Emantor — APPROVED** [review]
Tested on NixOS unstable.

File: pkgs/data/fonts/line-awesome/default.nix
Comment (timokau) [inline pkgs/data/fonts/line-awesome/default.nix:21]:
```suggestion
    description = "Replace Font Awesome with modern line icons";
```

https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/data/fonts/line-awesome/default.nix
Comment (timokau) [inline pkgs/data/fonts/line-awesome/default.nix:23]: No need to repeat the short description.

File: pkgs/data/fonts/line-awesome/default.nix
Comment (timokau) [inline pkgs/data/fonts/line-awesome/default.nix:15]: Why is this necessary? Shouldn't `fetchzip` take care of the unpacking?

Comment (puzzlewolf) [reply pkgs/data/fonts/line-awesome/default.nix:15]: Hm. I modeled the PR after #78409 and other fonts do the same thing, e.g. [yanone-kaffeesatz](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/yanone-kaffeesatz/default.nix) or [ubuntu-fonts](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/ubuntu-font-family/default.nix). But that doesn't mean it's correct :thinking: 

I think this overrides the `postFetch` defined by [`fetchzip`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchzip/default.nix), which is where the unpacking takes place. So it has to be done manually. But I concur, it's weird.

Comment (timokau) [reply pkgs/data/fonts/line-awesome/default.nix:15]: Yeah, it seems very weird. If we're unzipping manually anyway, we might as well use `fetchurl`.

Maybe this should use `mkDerivation` instead, disable the unnecessary phases (mostly build) and do the installation in `installPhase`. I don't have much experience with fonts in nixpkgs though. @marsam, what are your thoughts on this?

Comment (timokau) [reply pkgs/data/fonts/line-awesome/default.nix:15]: (Its possible I won't be able to react to emails over the weekend. So if someone else wants to merge, no need to wait for my approval)

Comment (doronbehar) [reply pkgs/data/fonts/line-awesome/default.nix:15]: @puzzlewolf according to the docs, you should just add `unzip` to `nativeBuildInputs`, see: https://nixos.org/nixpkgs/manual/#ssec-unpack-phase .

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/line-awesome/default.nix
Comment (puzzlewolf) [reply pkgs/data/fonts/line-awesome/default.nix:15]: @doronbehar Do you want me to model this after e.g. [nafees](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/nafees/default.nix)? It's quite different from e.g. [ubuntu-fonts](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/ubuntu-font-family/default.nix), it uses `mkDerivation` instead of `fetchzip`. Do you know of a community-agreed default way of doing this kind of font packages?

Comment (doronbehar) [reply pkgs/data/fonts/line-awesome/default.nix:15]: I can see that we have many fonts installed with `fetchzip` which is a bit unfortunate since this has the disadvantage that it makes it impossible to override these derivations with `<font>.overrideAttrs`. [nafees](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/nafees/default.nix) is written great so yes, please.

Comment (puzzlewolf) [reply pkgs/data/fonts/line-awesome/default.nix]: Done, now it uses mkDerivation.

**Comment by puzzlewolf** [comment]
And now it's fixed-output. It should be, right? The manual seems pretty clear: https://nixos.org/nix/manual/#sec-advanced-attributes.

File: pkgs/data/fonts/line-awesome/default.nix
Comment (timokau) [inline pkgs/data/fonts/line-awesome/default.nix:4]: You can use the normal `version` and `pname` split within `stdenv.mkDerivation` now.

**Review by timokau — CHANGES_REQUESTED** [review]
Looks much cleaner now!

It shouldn't be fixed output though. Fixed output is a tool that should only be used when needed, as it has many downsides (as you noted yourself). It isn't needed in this case.

**Review by timokau — APPROVED** [review]
Great! Thanks a lot for the contribution, your receptiveness to feedback and your patience.

**Comment by timokau** [comment]
If there's anything else you would like me to take a look at (a PR of yours or one you have reviewed), let me know. No guarantees of course.

**Comment by puzzlewolf** [comment]
Thanks for reviewing, explaining and your receptiveness on discourse :rocket:. Currently, there is only #83028, which should probably also go the  `mkDerivation` way.

I wonder if we should document somewhere in the manual that `mkDerivation` should be preferred over `fetchzip` for font packages. What do you think?

**Comment by timokau** [comment]
Yes, I think #83028 could be modeled after this PR. Documentation would be great! Are you up to writing that? I think the ´fetchX` functions are pretty much just intended to be used as part of a package, but never as *the* top-level package.

If you feel up to it, you could write some documentation. We can then look for some people who have contributed `fetchzip` derivations in the past and ping them, as they might have had reasons for that (though I can't think of any). After giving them some time to respond, we could go forward and hopefully have one less "gray area" in nixpkgs :)

**Pull request merged**
